### PR TITLE
OS.hpp removed as unused

### LINKED ISSUE #659: build error in python-alpine
I was trying to build htm.core using the python installation instructions. But I encountered some errors below. The build was executed in;

docker image: python-alpine
python: 3.7.4
gcc: 8.3.0

Am I missing something?

```
[ 17%] Building CXX object src/CMakeFiles/LibrarySource.dir/htm/ntypes/Scalar.cpp.o
[ 18%] Building CXX object src/CMakeFiles/LibrarySource.dir/htm/ntypes/Value.cpp.o
[ 18%] Building CXX object src/CMakeFiles/LibrarySource.dir/htm/os/Directory.cpp.o
[ 19%] Building CXX object src/CMakeFiles/LibrarySource.dir/htm/os/Env.cpp.o
[ 20%] Building CXX object src/CMakeFiles/LibrarySource.dir/htm/os/OS.cpp.o
[ 21%] Building CXX object src/CMakeFiles/LibrarySource.dir/htm/os/OSUnix.cpp.o
/htm.core/src/htm/os/OSUnix.cpp: In static member function 'static std::__cxx11::string htm::OS::getErrorMessageFromErrorCode(int)':
/htm.core/src/htm/os/OSUnix.cpp:87:30: error: invalid conversion from 'int' to 'char*' [-fpermissive]
   char *result = ::strerror_r(errorCode, errorBuffer, sizeof(errorBuffer));
                  ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/LibrarySource.dir/build.make:417: src/CMakeFiles/LibrarySource.dir/htm/os/OSUnix.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:144: src/CMakeFiles/LibrarySource.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
setup.py: Calling /htm.core/bindings/py/packaging/setup.py

Python version: 3.7.4 (default, Aug 21 2019, 00:19:59)
[GCC 8.3.0]

Traceback (most recent call last):
  File "setup.py", line 53, in <module>
    REPO_DIR = os.path.abspath(os.path.join(PY_BINDINGS, os.pardir, os.pardir, os.pardir))
  File "/usr/local/lib/python3.7/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/usr/local/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/htm.core/bindings/py/packaging/setup.py", line 312, in <module>
    getExtensionFiles(platform, build_type)
  File "/htm.core/bindings/py/packaging/setup.py", line 204, in getExtensionFiles
    generateExtensions(platform, build_type)
  File "/htm.core/bindings/py/packaging/setup.py", line 297, in generateExtensions
    subprocess.check_call(["cmake", "--build", ".", "--target", "install", "--config", build_type])
  File "/usr/local/lib/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install', '--config', 'Release']' returned non-zero exit status 2.
ERROR: Service 'app' failed to build: The command '/bin/sh -c apk add git && apk add cmake && apk add make && apk add g++                 && git clone https://github.com/htm-community/htm.core                 && cd htm.core && python setup.py install --user --force --prefix=' returned a non-zero code: 1
```

### PR DESCRIPTION
**Pull request by breznak**
and the code caused problems on Alpine linux.

This header has been (incorrectly) included in many files,
but is not needed.
Removes `getMemoryUsage` method which has been exported to python,
but apparently also never used.

Fixes #659 
For #175

### CODE REVIEW

File: bindings/py/cpp_src/bindings/engine/py_OS.cpp [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-namespace py = pybind11;
-
-using namespace htm;
-
-namespace htm_ext {
-
-    void init_OS(py::module& m)
-    {
-        py::class_<OS> py_OS(m, "OS");
-        py_OS.def_static("getProcessMemoryUsage", &htm::OS::getProcessMemoryUsage);
```
Comment (breznak) [inline bindings/py/cpp_src/bindings/engine/py_OS.cpp]: this is the only function that could have been somewhat useful, but even that is never used.

File: src/test/unit/os/DirectoryTest.cpp [hunk 2]
```diff
@@ -51,11 +51,11 @@ static std::string getCurrDir() {
     char buff[PATH_MAX];
 #if defined(NTA_OS_WINDOWS)
     DWORD res = ::GetCurrentDirectoryA(PATH_MAX - 1, (LPSTR)buff);
-    NTA_CHECK(res > 0) << OS::getErrorMessage();
+    NTA_CHECK(res > 0) << "Failed with " << res;
 
 #else
     char * s = ::getcwd(buff, PATH_MAX - 1);
-    NTA_CHECK(s != nullptr) << OS::getErrorMessage();
+    NTA_CHECK(s != nullptr) << "getcwd failed!";
```
Comment (breznak) [inline src/test/unit/os/DirectoryTest.cpp]: simplified behavior, but only in test and I don't think it's worth keeping that functionality.

File: bindings/py/cpp_src/bindings/engine/py_OS.cpp (hunk 1 above)
Comment (ctrl-z-9000-times) [reply bindings/py/cpp_src/bindings/engine/py_OS.cpp]: Python already has this in the standard library, albeit for Unix only:
https://docs.python.org/3.6/library/resource.html

Comment (breznak) [reply bindings/py/cpp_src/bindings/engine/py_OS.cpp]: Cool, I thought I've seen it used in your param-optimizer in python. 
Our "custom" code worked only on UNIX too, so no problem getting rid of it.

**Comment by breznak** [comment]
Nice compilation of the removed methods, and thanks for the review!

TODO: 
possible candidates for further removal: `os/Path,Directory,Env.hpp` 
In python we have a library for that, if c++ can avoid the out-of-scope code too (?)

**Comment by dkeeney** [comment]
> possible candidates for further removal: os/Path,Directory,Env.hpp

Path and Directory can be removable when all compilers support std::filesystem.

**Pull request merged**